src/lib/ptree/ptree.lua

```diff
@@ -343,6 +350,7 @@ function Manager:monitor_worker_counters(id)
    for ev in events.get, events do
       if has_suffix(ev.name, '.counter') then
          local name = strip_prefix(ev.name, dir..'/')
+         if blacklisted(name) then goto continue end
          local qualified_name = '/'..pid..'/'..name
          local counters = self.counters[name]
          if ev.kind == 'creat' then
```

Comment:
consider moving the blacklisted test here, i.e.

```lua
if blacklisted(name) then
   -- pass
elseif ev.kind == 'creat' then
...
```

just to avoid the goto